src/App.tsx

```diff
@@ -15,47 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET,
+  LENGTH,
+  NONE,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderOptions(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good, nextGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good.localeCompare(nextGood);
+
+      case SortType.LENGTH:
+        return good.length - nextGood.length;
+
+      case SortType.NONE:
```

Comment:
It's better to check this with an if statement so you start a sort loop only if you have to

Reply:
@lefteris-L I'd say that switch works better here. Limited number of conditions and we know them all exactly